### PR TITLE
Visitor can see list of article API

### PR DESCRIPTION
**Pull request by ViktorHek**
[Visitor can view all listed articles -API](https://www.pivotaltracker.com/story/show/176035762)
### User Story
```
As a Visitor
In order to choose an article to read
I would like to be able to see the article titles listed on a page
```
## Changes proposed in this pull request:
* writing request test for index action 
* creating index action 
* adding index in routes

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -11,6 +11,11 @@ def create
     end
   end
 
+  def index
+    articles = Article.all
+    render json: articles
```
Comment (emtalen) [inline app/controllers/api/articles_controller.rb]: use a serializer and also you need to return an array wrapped in a keyword, this is not restful

File: spec/requests/api/article_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    end
+
+    it "is expected to return 200 status" do
+      expect(response).to have_http_status 200
+    end
+
+    it "is expected to return a collection of articles" do
+      expect(response_json.count).to eq 5
+    end
+  end
```
Comment (emtalen) [inline spec/requests/api/article_index_spec.rb:34]: please extend the tests where you check that you get all the information you want from each article, the title, content and so on

**Review by emtalen — APPROVED** [review]

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (ViktorHek) [reply app/controllers/api/articles_controller.rb]: we have done that now :)

File: spec/requests/api/article_index_spec.rb (hunk 2 above)
Comment (ViktorHek) [reply spec/requests/api/article_index_spec.rb:34]: we have now added code in the test to test for author, title, subtitle and content

File: spec/requests/api/article_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    it "is expected to return 200 status" do
+      expect(response).to have_http_status 200
+    end
+
+    it "is expected to return a collection of articles" do
+      expect(response_json["articles"].count).to eq 5
+      expect(response_json["articles"][0]["author"]).to eq journalist.email
+      expect(response_json["articles"][0]["title"]).to eq "MyString"
+      expect(response_json["articles"][0]["sub_title"]).to eq "MyText"
+      expect(response_json["articles"][0]["content"]).to eq "MyText"
```
Comment (emtalen) [inline spec/requests/api/article_index_spec.rb:32]: you should only have one assertion in every `it` block, please fix that and divide it into several `it` blocks

**Review by emtalen — COMMENTED** [review]
Just fix the structure of the test and then this one is good to go

File: spec/requests/api/article_index_spec.rb (hunk 3 above)
Comment (ViktorHek) [reply spec/requests/api/article_index_spec.rb:32]: we are done now

**Review by emtalen — APPROVED** [review]

**Pull request merged**